File: src/main/java/me/kuku/yuq/entity/WeiboEntity.java
package me.kuku.yuq.entity;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONArray;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Entity
@Table(name = "weibo")
@Data
@AllArgsConstructor
@NoArgsConstructor
public class WeiboEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    @Column(unique = true)
    private Long qq;
    @Column(name = "group_")
    private Long group;
    @Lob
    @Column(columnDefinition="text")
    private String pcCookie;
    @Lob
    @Column(columnDefinition="text")
    private String mobileCookie;
    private Boolean monitor;
    @Lob
    @Column(columnDefinition="text")
    private String likeList;
    @Lob
    @Column(columnDefinition="text")
    private String commentList;
    @Lob
    @Column(columnDefinition="text")
    private String forwardList;

    public WeiboEntity(String pcCookie, String mobileCookie){
        this.pcCookie = pcCookie;
        this.mobileCookie = mobileCookie;
    }

    public WeiboEntity(Long qq, Long group){
        this.qq = qq;
        this.group = group;
    }

    public JSONArray getLikeJsonArray(){
        if (likeList == null) return new JSONArray();
        else return JSON.parseArray(likeList);
    }

    public void setLikeJsonArray(JSONArray jsonArray){
        this.likeList = jsonArray.toString();
    }

    public JSONArray getCommentJsonArray(){
        if (commentList == null) return new JSONArray();
        else return JSON.parseArray(commentList);
    }

    public void setCommentJsonArray(JSONArray jsonArray){
        this.commentList = jsonArray.toString();
    }

    public JSONArray getForwardJsonArray(){
        if (forwardList == null) return new JSONArray();
        else return JSON.parseArray(forwardList);
    }

    public void setForwardJsonArray(JSONArray jsonArray){
        this.forwardList = jsonArray.toString();
    }
}

File: src/main/kotlin/me/kuku/yuq/entity/WeiboEntity.kt
package me.kuku.yuq.entity

import org.hibernate.annotations.Type
import org.springframework.data.jpa.repository.EntityGraph
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Service
import javax.persistence.*

@Entity
@Table(name = "weibo")
@NamedEntityGraph(name = "weibo_qq", attributeNodes = [NamedAttributeNode(value = "qqEntity", subgraph = "qqEntity")],
    subgraphs = [NamedSubgraph(name = "qqEntity", attributeNodes = [NamedAttributeNode("groups")])])
class WeiboEntity: BaseEntity() {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Int? = null
    @OneToOne
    @JoinColumn(name = "qq_id")
    var qqEntity: QqEntity? = null
    @Column(length = 2000)
    var pcCookie: String = ""
    @Column(length = 2000)
    var mobileCookie: String= ""
    @Type(type = "json")
    @Column(columnDefinition = "text")
    var config: WeiboConfig = WeiboConfig()
}

interface WeiboRepository: JpaRepository<WeiboEntity, Int> {
    fun findByQqEntity(qqEntity: QqEntity): WeiboEntity?

    @EntityGraph(value = "weibo_qq", type = EntityGraph.EntityGraphType.FETCH)
    override fun findAll(): List<WeiboEntity>
}

@Service
class WeiboService (
    private val weiboRepository: WeiboRepository
) {

    fun save(weiboEntity: WeiboEntity): WeiboEntity = weiboRepository.save(weiboEntity)

    fun findAll(): List<WeiboEntity> = weiboRepository.findAll()

    fun findByQqEntity(qqEntity: QqEntity) = weiboRepository.findByQqEntity(qqEntity)

}

data class WeiboConfig(var push: Status = Status.OFF, var sign: Status = Status.OFF)